lib/MusicBrainz/Server/Controller/Account.pm

```diff
@@ -752,6 +752,17 @@ sub revoke_application_access : Path('/account/applications/revoke-access') Args
     my ($self, $c, $application_id, $scope) = @_;
 
     my $form = $c->form( form => 'Confirm' );
+
+    my $token_exists = $c->model('EditorOAuthToken')->check_granted_token(
+        $c->user->id,
+        $application_id,
+        $scope,
+    );
+    $c->detach(
+        '/error_404',
+        [ l('There is no OAuth token with these parameters.') ]
```

Comment:
About the commit message rather than this line of code:

> the only way a user should get here anyway is if they edit the URL by hand

or if they reopened a tab revoking an already revoked token.